Tmedia 734 gallery block update

JackHowa opened the pull request:
## Description

Update gallery for v2 architecture and images

## Jira Ticket

- [TMEDIA-734](https://arcpublishing.atlassian.net/browse/TMEDIA-734)

## Acceptance Criteria

Acceptance Criteria:

Gallery block is migrated to the new architecture.

## Test Steps

## Chromatic 

1. Open chromatic diff 
2. See the carousel with no content passed in matching the figma mocks with spacing and font-size 
3. Open the carousel with content passed in to see the image full width and 16:9 based on the mocks 
4. See the media item hide and show credits, caption and title 

## Live 

1. Checkout this branch `git checkout TMEDIA-734-gallery-block-update`
2. Checkout the feature pack `git checkout TMEDIA-734-gallery-block-update`
3. Use the themes internal env
4. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/gallery-block,@wpmedia/global-phrases-block`
5. Create a page 
6. Set the global content with "/2022/08/30/gallery-title/" for the content-api for a gallery 
7. Add the block Gallery
8. Ensure that it is inheriting global content

## Dependencies or Side Effects

- discrepancies noted for the caption and black area around the image. Using approach that respects aspect ratio like before
- manifest https://github.com/WPMedia/arc-themes-manifests/pull/129
- Feature pack https://github.com/WPMedia/arc-themes-feature-pack/pull/341
- In the chromatic the height of the top bar is different. Normally we won't have that long of a string or without breaks. I looked into constraining the height. We may need to implement the height and line clamp constriction on the carousel potentially if this is a major concern. I deferred it for now.

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

# Todo 
- [x] Update content "/2022/08/30/gallery-title/" to have caption, title, credits for testing purpose -> updated first pic in the gallery, waiting for change to update in content source. I see it in the debugger
- [x] Match spacing of the top right buttons to the figma mocks 
- [x] Check if the auth token check is necessary ternary
- [x] Match new article body implementation with black bars and respecting aspect ratio

# Notes 

- I don't see the auth object on the content-api. Because it hasn't been implemented with the signing service yet, right? Is this the best course of action without mock data? I looked and found a gallery inside the themes internal (in test steps). I noticed with the content-api content source that it hasn't been implemented with the signing service -- still has the v1. -> fixed with Matthew suggestion for content source "/2022/08/30/gallery-title/"
- I'm investigating the sr-only. Even if we make it `visually-hidden` (per the reset styles), it would still be shown because it's actually being rendered in the html for the additional steps. this should be dangerously set right? Then we will need to also adjust the global translations. -> updated translations and the components
<img width="687" alt="Screen Shot 2022-08-29 at 17 45 44" src="https://user-images.githubusercontent.com/5950956/187312322-d31b4408-c89a-4166-a1e7-a51237930d94.png">
- I'm getting a warning for the ad element not passing in for types. Need to investigate how to fix the warning and/or ensure that the type is a react element as expected -> Matthew fixed
- In terms of v2 design, the full-width image seems to be the goal now. I added this to a discrepancy. Originally, I was mimicking a gallery that was unintentionally full-width in the article gallery. But that was fixed in a related pr https://github.com/WPMedia/arc-themes-blocks/pull/1494

JackHowa committed:
chore: Remove unused index.js file. Update documentation for package json per v2 refactor checklist

JackHowa committed:
chore: Remove unused npmignore for mocks

JackHowa committed:
docs: Update props to include lazy load

JackHowa committed:
docs: Add other fields for hiding and showing the image title

JackHowa committed:
feat: Add basic container for carousel with no content

JackHowa committed:
feat: Hide additional controls on mobile

JackHowa committed:
Merge branch 'arc-themes-release-version-2.0.1' into TMEDIA-734-gallery-block-update

JackHowa committed:
chore: Update theme components to fix prev and next button bug in gallery

JackHowa committed:
fix: Update padding to match designs and font

JackHowa committed:
feat: Update line height and weight to be body tiny

JackHowa committed:
feat: Add carousel track button color update using block styles per mocks and v1

JackHowa committed:
feat: Add carousel control bottom padding between controls and track

JackHowa committed:
feat: Add caption and title handling

JackHowa committed:
fix: Remove background color for now

JackHowa committed:
test: Fix react hooks errors. Internationalization import update. es6 import update

JackHowa committed:
feat: Add image resizing for the gallery mocks

JackHowa committed:
test: Add basic enzyme setup with working mocks

JackHowa committed:
test: Remove unnecessary logic. Update tests to check the props of the helper theme components

JackHowa committed:
Merge branch 'arc-themes-release-version-2.0.1' into TMEDIA-734-gallery-block-update

JackHowa committed:
fix: Add optional chaining if no auth key

matthewroach commented on blocks/gallery-block/features/gallery/default.jsx:
```diff
@@ -58,30 +71,86 @@ export const GalleryPresentation = ({
+					<Icon name="Close" />
+				</button>
 			}
 			adElement={/* istanbul ignore next */ () => <AdBlock />}
```
The react warning you are seeing for `adElement` is because you are passing a function as the prop this should be 
```suggestion
			adElement={/* istanbul ignore next */ <AdBlock />}
```

JackHowa replied:
updated by you and looks good

matthewroach committed:
Update gallery story data to be correct

matthewroach committed:
Fix adElement warning and update image src value

JackHowa committed:
fix: Add id to update warning for carousel handling of buttons and autoplay

JackHowa committed:
fix: Ensure next and previous arrows and next to each other rather than atop

JackHowa committed:
fix: Use themes components class for screen reader. Update translations in lokalise and generate result from there

Commit:
Merge branch 'arc-themes-release-version-2.0.1' into TMEDIA-734-gallery-block-update

JackHowa committed:
docs: Add third story to see both arrows direction

JackHowa committed:
chore: Update theme components to get translations inner html

JackHowa committed:
fix: Remove additional padding. Match figma designs for buttons

JackHowa committed:
Merge branch 'TMEDIA-734-gallery-block-update' of github.com:WPMedia/fusion-news-theme-blocks into TMEDIA-734-gallery-block-update

JackHowa committed:
feat: Add background black bars based on previous article body implementation

JackHowa committed:
refactor: Remove auth update

JackHowa committed:
style: Update spacing

JackHowa commented on blocks/gallery-block/_index.scss:
```diff
@@ -0,0 +1,14 @@
+		@include scss.block-components("gallery-track-icon");
+		@include scss.block-properties("gallery-track-icon");
+	}
+	&__image-wrapper {
```
fixed

JackHowa committed:
feat: Add back carousel actions as the buttons are visible in the figma mocks for mobile

Commit:
Merge branch 'arc-themes-release-version-2.0.1' into TMEDIA-734-gallery-block-update

matthewroach commented on .storybook/themes/news.scss:
```diff
@@ -911,6 +911,63 @@
+							padding: 0 var(--global-spacing-4),
+						),
+						carousel-button-next: (
+							background-color: transparent,
```
This is causing a regression in other blocks - we will need to define the default at the component token level

JackHowa replied:
updated, my bad

matthewroach commented on .storybook/themes/news.scss:
```diff
@@ -911,6 +911,63 @@
+							background-color: transparent,
+						),
+						carousel-button-previous: (
+							background-color: transparent,
```
This is causing a regression in other blocks - we will need to define the default at the component token level

JackHowa replied:
updated, my bad. should've looked at the whole regression

matthewroach commented on blocks/gallery-block/features/gallery/default.jsx:
```diff
@@ -58,30 +72,89 @@ export const GalleryPresentation = ({
-					phrases.t("global.gallery-page-count-text", { current, total })
+			enableFullScreen
+			fullScreenShowButton={
+				<button type="button">
```
`button` requires an `id` - and an `id` must be unique

JackHowa replied:
updated

matthewroach replied:
I miss spoke on the `id` value here - the id's of the buttons should match the `id` prop of the `Carousel`

JackHowa replied:
id is not needed here. the aria controls should match

matthewroach commented on blocks/gallery-block/features/gallery/default.jsx:
```diff
@@ -58,30 +72,89 @@ export const GalleryPresentation = ({
+				</button>
+			}
+			fullScreenMinimizeButton={
+				<button type="button">
```
`button` requires an `id` - and an `id` must be unique

JackHowa replied:
updated

JackHowa replied:
will update article body as well

JackHowa replied:
https://github.com/WPMedia/arc-themes-blocks/pull/1497 article body gallery updated to fix

matthewroach replied:
I miss spoke on the `id` value here - the id's of the buttons should match the `id` prop of the `Carousel`

JackHowa replied:
the aria controls are on the button crucially

JackHowa replied:
![Screen Shot 2022-08-31 at 10 15 55](https://user-images.githubusercontent.com/5950956/187714905-bc094090-fde4-4400-9668-a26987942ea6.png)

JackHowa replied:
we shouldn't have duplicate ids though

nschubach replied:
The Carousel.Button "id" is not the HTML id property... it's poorly named I think.

![image](https://user-images.githubusercontent.com/2287238/187788069-cb66835e-1853-44a0-ab53-9226c376cc08.png)

matthewroach commented on blocks/gallery-block/features/gallery/default.jsx:
```diff
@@ -58,30 +72,89 @@ export const GalleryPresentation = ({
+			adElement={/* istanbul ignore next */ <AdBlock />}
+			adInterstitialClicks={interstitialClicks}
+			nextButton={
+				<Carousel.Button id={id} label="Next Slide">
```
`id` must be unique

JackHowa replied:
updated

matthewroach replied:
I miss spoke on the `id` value here - the id's of the buttons should match the `id` prop of the `Carousel`

JackHowa replied:
id matches here correctly

matthewroach commented on blocks/gallery-block/features/gallery/default.jsx:
```diff
@@ -58,30 +72,89 @@ export const GalleryPresentation = ({
+				</Carousel.Button>
+			}
+			previousButton={
+				<Carousel.Button id={id} label="Previous Slide">
```
`id` must be unique

JackHowa replied:
updated -- my bad. I'll work on updating the article body button logic as well. I wrongly assumed something else was happening here with id

matthewroach replied:
I miss spoke on the `id` value here - the id's of the buttons should match the `id` prop of the `Carousel`

JackHowa replied:
updated

matthewroach commented on blocks/gallery-block/features/gallery/default.jsx:
```diff
@@ -58,30 +72,89 @@ export const GalleryPresentation = ({
+								resizerURL={resizerURL}
+								resizedOptions={{ auth: galleryItem.auth[resizerAppVersion] }}
+								// 16:9 aspect ratio
+								width={375}
```
Seems a little small for the sizes we are outputting - we are saying the image is always `100vw` - What do you think of using the 500 size here?

JackHowa replied:
500 width here?

JackHowa replied:
updated the width to be wider. also reduced 100vh based off of 75vh max

JackHowa committed:
Merge branch 'TMEDIA-734-gallery-block-update' of github.com:WPMedia/fusion-news-theme-blocks into TMEDIA-734-gallery-block-update

JackHowa committed:
fix: Update defaults due to side effect of nested components

JackHowa committed:
docs: Fix mock data with duplicate id no longer here

JackHowa committed:
fix: Update ids based on feedback for full screen

via https://github.com/WPMedia/arc-themes-blocks/pull/1491#discussion_r959605253

JackHowa committed:
fix: Update to use unique ids for buttons

JackHowa committed:
fix: Update to use smaller epected vw based off of feedback and 75 vh expectation

JackHowa commented:
Feedback updated based on reviews @matthewroach

JackHowa committed:
Revert "fix: Update ids based on feedback for full screen"

This reverts commit 83e8e96100799eeec92f7c08164aa26447a19dfb.

JackHowa committed:
Revert "fix: Update to use unique ids for buttons"

This reverts commit fc6ec52ebb88d649163e88ea54e4194ea84b2a4d.

nschubach commented on blocks/gallery-block/features/gallery/default.jsx:
```diff
@@ -35,7 +48,7 @@ export const GalleryPresentation = ({
 	}
 
-	const { resizerURL, galleryCubeClicks, locale = "en" } = getProperties(arcSite);
+	const { galleryCubeClicks, locale } = getProperties(arcSite);
```
no default?  This will cause issues won't it?
```suggestion
	const { galleryCubeClicks, locale = "en" } = getProperties(arcSite);
```

JackHowa replied:
locale is en by default in fusion as far as I know

nschubach replied:
I thought it was confirmed that it wasn't because fusion didn't want to default it.  If that's the case, I'm happy about that... but I had heard different.  @matthewroach: do you know?

matthewroach replied:
We can assume it's going to be passed as `en` as at minimum the default value in properties should be set via themes settings

nschubach replied:
so we can start leaving this off?

nschubach commented on blocks/gallery-block/features/gallery/default.jsx:
```diff
@@ -58,30 +72,91 @@ export const GalleryPresentation = ({
+										isDefault: true,
+										// 75vh max height
+										// if 16/9 aspect ratio, then roughly 40vw expected
+										sourceSizeValue: "40vw",
```
Oof, I don't have a solve for this, but that value will be different on mobile because of the flipped aspect ratio.

nschubach replied:
480px:
![image](https://user-images.githubusercontent.com/2287238/187774059-3b4b0cb0-314e-43cd-86e0-f79b61e35aeb.png)
 :/

nschubach replied:
I see you changed the width on this... double checking

JackHowa replied:
ya it was looking weirdly without it. I think 100vw might be ok if we constrain max height

nschubach replied:
Actually, since you only have one size, there's no point in including the sizes array,  Removing it changes nothing as far as I can tell here.
![image](https://user-images.githubusercontent.com/2287238/187776760-bcfe0e52-7c42-40a0-b955-afa65a27c081.png)

nschubach replied:
I think it would be useful if you had a media query size:  https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/sizes#examples

nschubach replied:
I assume removing it would be that 100vw you were referring to

nschubach replied:
not going to sweat this... from what I understand of it, it might be a good plan to pass it the size of the container it's in so it can properly determine the image to use

JackHowa replied:
It's removed -- thanks

nschubach commented on blocks/gallery-block/package.json:
```diff
@@ -20,14 +19,10 @@
   },
   "gitHead": "83fb8d6685958d85ddbfcfbf01d9a5c864cd6c95"
-}
+}
```
```suggestion
}

```

JackHowa committed:
feat: Increase the vw to ensure that the image grows vertically

nschubach reviewed (commented):
Questions for now... still looking

Commit:
Update blocks/gallery-block/package.json

Co-authored-by: nschubach <nschubach@gmail.com>

Commit:
Merge branch 'arc-themes-release-version-2.0.1' into TMEDIA-734-gallery-block-update

ecarlisle commented:
@JackHowa : Seeing the following locally on load. I followed all steps on the "Live" setup above. Wanted to see if you're seeing the same.
<img width="1213" alt="image" src="https://user-images.githubusercontent.com/452134/187777315-1f6b3823-b39b-477b-8ac9-6c3ed8938415.png">

ecarlisle commented:
Also seeing this error when using expand button.
<img width="954" alt="image" src="https://user-images.githubusercontent.com/452134/187777854-a298c0e5-16f4-4420-875d-bca36625d97b.png">

nschubach commented:
> @JackHowa : Seeing the following locally on load. I followed all steps on the "Live" setup above. Wanted to see if you're seeing the same. <img alt="image" width="1213" src="https://user-images.githubusercontent.com/452134/187777315-1f6b3823-b39b-477b-8ac9-6c3ed8938415.png">

I'm pretty sure this is a pagebuilder bug.  I was getting something similar in the footer when I didn't specify the Hierarchy field on the content source and it was throwing me for a loop until I put a value in for it.

JackHowa commented:
> Also seeing this error when using expand button. <img alt="image" width="954" src="https://user-images.githubusercontent.com/452134/187777854-a298c0e5-16f4-4420-875d-bca36625d97b.png">

Huh interesting. I'll retry with live but I'm not seeing it in chromatic https://5f91e4721ccaba0022a10782-mpllxnsbbx.chromatic.com/iframe.html?id=blocks-gallery--with-passed-in-content&viewMode=story


https://user-images.githubusercontent.com/5950956/187778822-ba8888ae-2609-4c95-93db-a3f7e6732417.mov

JackHowa commented:
> > @JackHowa : Seeing the following locally on load. I followed all steps on the "Live" setup above. Wanted to see if you're seeing the same. <img alt="image" width="1213" src="https://user-images.githubusercontent.com/452134/187777315-1f6b3823-b39b-477b-8ac9-6c3ed8938415.png">
> 
> I'm pretty sure this is a pagebuilder bug. I was getting something similar in the footer when I didn't specify the Hierarchy field on the content source and it was throwing me for a loop until I put a value in for it.

yeah even when I provide fallbacks, I'm seeing issues. restarting fusion to sanity check

nschubach commented:
I am not getting those errors...
![image](https://user-images.githubusercontent.com/2287238/187784181-8465bd4d-e6dd-4b0f-919f-a49e0bdfc46c.png)

JackHowa commented:
> I am not getting those errors...

using global content

nschubach commented:
> > I am not getting those errors...
> 
> using global content

Even if I use the block content:
![image](https://user-images.githubusercontent.com/2287238/187785296-a72be503-f183-428d-8e3a-cbb148bcd19f.png)

JackHowa committed:
fix: Use _id rather than id for gallery fix showing full screen

JackHowa committed:
Merge branch 'TMEDIA-734-gallery-block-update' of github.com:WPMedia/fusion-news-theme-blocks into TMEDIA-734-gallery-block-update

nschubach commented:
I do get the request fullscreen error when viewing the published page unfortunately when trying to expand it:
![image](https://user-images.githubusercontent.com/2287238/187786280-c55f1472-65d5-46f7-965f-24712e19ecbe.png)
looking...

JackHowa committed:
fix: Remove sizes media query

JackHowa commented:
@nschubach just fixed the full screen issue. interestingly, this issue only shows in production. as before I was getting the wrong id

JackHowa commented:
wonder if it's a fusion release 3.3.1 bug

JackHowa commented:
@ecarlisle thanks for the review. I've fixed the fill screen. interestingly, it wasn't showing on storybook. now it's working on sb and live

JackHowa committed:
fix: Update mock global content to update the storybook id issue

nschubach reviewed (approved):
I'm happy with what I see right now... everything seems to work with the exception of what we believe is a fusion bug (3.3.0 is fine)

JackHowa committed:
fix: Use tiny font for gallery based on dqa

JackHowa committed:
fix: Ensure the article body has a background color of transparent for maintaining previous implementation

JackHowa commented:
approved by kelly -- it was moved to my test queue

JackHowa committed:
Merge branch 'arc-themes-release-version-2.0.2' into TMEDIA-734-gallery-block-update
